astrodb_utils/utils.py

```diff
@@ -30,59 +30,129 @@ class AstroDBError(Exception):
         Path to data directory; default 'data/'
     recreatedb : bool
         Flag whether or not the database file should be recreated
     reference_tables : list
         List of tables to consider as reference tables.
         Default: Publications, Telescopes, Instruments, Versions, PhotometryFilters
+        Looks in <db_name>/__init__.py for the list of tables.
     felis_schema : str
         Path to Felis schema; default None
+        Looks in db_name/schema.yaml for the schema path.
+
+    Returns
+    -------
+    db : Astrodbkit Database object
+
+    Also creates the database file if it does not exist.
+    If recreatedb is True, it removes the existing database file before creating a new one.
     """
 
-    db_file_path = Path(db_file)
+    db_name = os.path.basename(db_path)
+
+    db_file = db_name + ".sqlite"
     db_connection_string = "sqlite:///" + db_file
+    logger.debug(f"Database connection string: {db_connection_string}")
+
+    # Load the reference tables from the db_name module if not provided
+    if reference_tables is None:
+        try:
+            init_path = os.path.join(db_path, "__init__.py")
+            spec = importlib.util.spec_from_file_location(db_name, init_path)
+            db_module = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(db_module)
+            REFERENCE_TABLES = db_module.REFERENCE_TABLES
+        except ImportError:
+            logger.warning(f"Could not import reference tables from {db_path}, using default set.")
+            REFERENCE_TABLES = [
+                "Publications",
+                "Telescopes",
+                "Instruments",
+                "Versions",
+                "PhotometryFilters",
+                "Regimes",
+                "AssociationList",
+                "ParameterList",
+                "CompanionList",
+                "SourceTypeList",
+            ]
+    else:
+        REFERENCE_TABLES = reference_tables
 
-    # removes the current .db file if one already exists
-    if recreatedb and db_file_path.exists():
+    # removes the current .db file if one already exists and Recreatedb is True
+    if recreatedb and os.path.exists(db_file):
         os.remove(db_file)
 
-    if not db_file_path.exists():
-        # Create database, using Felis if provided
-        create_database(db_connection_string, felis_schema=felis_schema)
-        # Connect and load the database
-        db = Database(db_connection_string, reference_tables=reference_tables)
-        if logger.parent.level <= 10:
-            db.load_database(data_path, verbose=True)
+    if not os.path.exists(db_file):
+        db = _rebuild_db(db_path, db_file, data_path, REFERENCE_TABLES, felis_schema)
+    else:
+        # if database file already exists, just connect to it
+        db = Database(db_connection_string, reference_tables=REFERENCE_TABLES)
+
+    return db
+
+
+def _rebuild_db(db_path, db_file, data_path, reference_tables, felis_schema):
```

Comment:
May want to set `felis_schema = None` here to avoid issues with line 122 below (which checks whether it's None or not)